### PR TITLE
split tweet messages

### PR DESCRIPTION
**Pull request by rangelrey**

### CODE REVIEW

File: hot_tweets_bot/app.py [hunk 1]
```diff
@@ -327,39 +335,39 @@ def callback_chrono_backup(self, context: CallbackContext):  # pragma: no cover
...
         for search in searches:
             if search:
                 tweets = self.twitter_service.fetch_top_n_tweets(search, n=3)
                 search_msg: str = messages.build_top_n_tweet_message(tweets, search)
                 search_msgs.append(search_msg)
 
-        searches_msg = "\n\n\n".join(search_msgs)
-        msg = f'{self.intro_send_tweets_msg}\n\n\n{searches_msg}'
-        return msg
+        search_msgs[0] = f"{self.intro_send_tweets_msg}\n\n\n{search_msgs[0]}"
```
Comment (rooom13) [inline hot_tweets_bot/app.py:354]: If there are no searches this will throw an exception. Pls change it + unit test all edge cases

**Review by rooom13 — APPROVED** [review]

File: hot_tweets_bot/messages.py [hunk 2]
```diff
@@ -56,5 +56,9 @@ def build_top_n_tweet_message(tweets, search):
         return f'\n\n<pre>🔎 No tweets found for {utils.highlight(search)} in the last 24h 🔍</pre>\n'
 
 
-def build_trial_fetch_msg(search: str, tweet_msg: str) -> str:
+def build_trial_fetch_msg(search: str, tweet_msg_list: str) -> str:
+    tweet_msg = ""
+    for msg in tweet_msg_list:
+        tweet_msg = tweet_msg + "\n" + msg
```
Comment (rooom13) [inline hot_tweets_bot/messages.py:62]: `tweet_msg += f"\n{msg}"` better :)

File: hot_tweets_bot/app.py [hunk 3]
```diff
@@ -327,41 +335,44 @@ def callback_chrono_backup(self, context: CallbackContext):  # pragma: no cover
...
+            for msg in message_list:
+                if msg:
+                    self.send_message_to_user(chat_id, msg)
 
     @staticmethod
     def is_admin(chat_id: str):
         return str(chat_id) in config.ADMIN_CHAT_IDS
 
-    def get_searches_msg(self, searches: List[str]):
+    def get_searches_msg_list(self, searches: List[str]):
```
Comment (rooom13) [inline hot_tweets_bot/app.py:348]: let's rather do the following:
````
def get_search_msg(self, search: str):
    ...

then it can be used in 

def get_searches_msg_list(searches: List[str]):

````

by done so, in line 372 you would use `get_search_msg` instead.

File: hot_tweets_bot/app.py [hunk 4]
```diff
@@ -327,41 +335,44 @@ def callback_chrono_backup(self, context: CallbackContext):  # pragma: no cover
...
     @handle_error_send_user
     def send_message_to_user(self, chat_id: str, msg: str, reply_markup=None):
         self.updater.bot.send_message(chat_id=chat_id, text=msg, parse_mode='HTML', reply_markup=reply_markup)
 
     @handle_error_send_user
     def send_trial_message_to_user(self, chat_id: str, search: str):  # pragma no cover
-        searches_msg = self.get_searches_msg([search])
-        tweet_msg = messages.build_trial_fetch_msg(search, searches_msg)
-        self.send_message_to_user(chat_id, tweet_msg)
+        searches_msg_list = self.get_searches_msg_list([search])
```
Comment (rooom13) [inline hot_tweets_bot/app.py:372]: here u would then use get_search_msg(search)

File: hot_tweets_bot/messages.py [hunk 5]
```diff
@@ -56,5 +56,9 @@ def build_top_n_tweet_message(tweets, search):
         return f'\n\n<pre>🔎 No tweets found for {utils.highlight(search)} in the last 24h 🔍</pre>\n'
 
 
-def build_trial_fetch_msg(search: str, tweet_msg: str) -> str:
+def build_trial_fetch_msg(search: str, tweet_msg_list: str) -> str:
```
Comment (rooom13) [inline hot_tweets_bot/messages.py:59]: why does it accept a list (tweet_msg_list) ? (Which by the way the type annotation (str) is wrong)

This method is only called with a single element (line 372 from app.py)

**Review by rooom13 — CHANGES_REQUESTED** [review]

File: tests/test_app.py [hunk 6]
```diff
@@ -56,13 +56,30 @@ def test_get_top_tweets_msg(mocked_dao, mocked_auth):
     app.dao.save_user(message)
     app.dao.save_user_search(chat_id, "music")
     dao.r.flushall(app.dao)
-    res = app.get_top_tweets_msg(chat_id)
-    assert res == '''<b>🔹
+    res = app.get_top_tweets_msg_list(chat_id)
+    assert res == ['''<b>🔹
 🔹 <pre>𝕐𝕠𝕦𝕣 𝕋𝕠𝕡 𝕋𝕨𝕖𝕖𝕥𝕤 - 𝕃𝕒𝕤𝕥 24𝕙</pre>
-🔹</b>
+🔹</b>''', '''<pre> -------------------
```
Comment (rooom13) [inline tests/test_app.py:62]: '''<pre> -------------------
needs to start in next line

File: tests/test_messages.py [hunk 7]
```diff
@@ -81,4 +81,4 @@ def func(a):
 def test_build_trial_fetch_msg():
     tweet_msg = "top tweets"
     search = "bitcoin"
-    assert f"These are the top 3 tweets for {search}:\n\n\n {tweet_msg}" == messages.build_trial_fetch_msg(search, tweet_msg)
+    assert f"These are the top 3 tweets for {search}:\n\n\n \n{tweet_msg}" == messages.build_trial_fetch_msg(search, tweet_msg)
```
Comment (rooom13) [inline tests/test_messages.py:84]: why the whitespace before the last \n

File: hot_tweets_bot/app.py [hunk 8]
```diff
@@ -327,41 +335,46 @@ def callback_chrono_backup(self, context: CallbackContext):  # pragma: no cover
...
+                if msg:
+                    self.send_message_to_user(chat_id, msg)
 
     @staticmethod
     def is_admin(chat_id: str):
         return str(chat_id) in config.ADMIN_CHAT_IDS
 
-    def get_searches_msg(self, searches: List[str]):
+    def get_search_msg(self, search: str):
+        if search:
```
Comment (rooom13) [inline hot_tweets_bot/app.py:349]: is this check need? if not raus

File: hot_tweets_bot/app.py [hunk 9]
```diff
@@ -327,41 +335,46 @@ def callback_chrono_backup(self, context: CallbackContext):  # pragma: no cover
 
     @handle_error_send_user
     def send_user_top_tweets(self, chat_id):  # pragma: no cover
```
Comment (rooom13) [inline hot_tweets_bot/app.py:337]: add type hint

File: hot_tweets_bot/app.py [hunk 10]
```diff
@@ -327,41 +335,46 @@ def callback_chrono_backup(self, context: CallbackContext):  # pragma: no cover
...
     def is_admin(chat_id: str):
         return str(chat_id) in config.ADMIN_CHAT_IDS
 
-    def get_searches_msg(self, searches: List[str]):
+    def get_search_msg(self, search: str):
+        if search:
+            tweets = self.twitter_service.fetch_top_n_tweets(search, n=3)
+            return messages.build_top_n_tweet_message(tweets, search)
+
+    def get_searches_msg_list(self, searches: List[str]):
```
Comment (rooom13) [inline hot_tweets_bot/app.py:353]: add return type hint

**Review by rooom13 — APPROVED** [review]
